Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -47,23 +47,24 @@ Status& Status::CausedBy(const Status& cause) {
     return *this;
   }
 
-  absl::StrAppend(&state_->msg, ": ", cause.error_message());
+  absl::StrAppend(&AssertStatePtr()->msg, ": ", cause.error_message());
```

Comment:
`util::Status` in google3 does not fail here. In general I think we should avoid failing. I've sent you notes privately about comparisons with google3 Status.

Reply:
Changed such that access after moved will not crash.